crates/cli/src/execute.rs

```diff
@@ -0,0 +1,36 @@
+use crate::transaction::CliTransactionArgs;
+use alloy_ethers_typecast::transaction::ExecutableTransaction;
+use alloy_sol_types::SolCall;
+use anyhow::Result;
+use clap::Args;
+use clap::FromArgMatches;
+use clap::Parser;
+use rain_orderbook_common::transaction::TransactionArgs;
+
+#[derive(Parser)]
+pub struct ExecutableTransactionCall<T: FromArgMatches + Args> {
```

Comment:
@mattyg this is cool, only thing i'd say is this abstraction will probably break when we get to more complex commands like add-order, remove-order, take-orders etc where the cli args don't match the contract call args

Reply:
This just abstracts the logic for making a call with a ledger client and given transaction cargs -- conversion from cli args to contract call is still implemented manually for each cli command in the Execute trait.